File: java/src/in3/btc/ScriptPubKey.java
package in3.btc;

import in3.utils.JSON;

/**
 * Script on a transaction output.
 */
public class ScriptPubKey {
  private JSON data;

  private ScriptPubKey(JSON data) {
    this.data = data;
  }

  protected static ScriptPubKey asScriptPubKey(Object o) {
    if (o == null)
      return null;
    return new ScriptPubKey((JSON) o);
  }

  protected static ScriptPubKey[] asScriptPubKeys(Object o) {
    if (o == null)
      return null;
    Object[] a       = (Object[]) o;
    ScriptPubKey[] b = new ScriptPubKey[a.length];
    for (int i = 0; i < a.length; i++)
      b[i] = ScriptPubKey.asScriptPubKey(a[i]);
    return b;
  }

  /**
   * The hash of the blockheader.
   */
  public String getAsm() {
    return data.getString("asm");
  }

  /**
   * The raw hex data.
   */
  public String getHex() {
    return data.getString("hex");
  }

  /**
   * The required sigs.
   */
  public long getReqSigs() {
    return data.getLong("reqSigs");
  }

  /**
   * The type e.g.: pubkeyhash.
   */
  public String getType() {
    return data.getString("type");
  }

  /**
   * List of addresses.
   */
  public String[] getAddresses() {
    return data.getStringArray("addresses");
  }
}

File: dotnet/In3/Btc/ScriptPubKey.cs
using System.Text.Json.Serialization;

namespace In3.Btc
{
    /// <summary>
    /// Script on a transaction output.
    /// </summary>
    public class ScriptPubKey
    {
        /// <summary>
        /// The asm data,
        /// </summary>
        [JsonPropertyName("asm")] public string Asm { get; set; }

        /// <summary>
        /// The raw hex data.
        /// </summary>
        [JsonPropertyName("hex")] public string Hex { get; set; }

        /// <summary>
        /// The required sigs.
        /// </summary>
        [JsonPropertyName("reqSigs")] public uint ReqSigs { get; set; }

        /// <summary>
        /// The type.
        /// </summary>
        /// <example>pubkeyhash</example>
        [JsonPropertyName("type")] public string Type { get; set; }

        /// <summary>
        /// List of addresses.
        /// </summary>
        [JsonPropertyName("addresses")] public string[] Addresses { get; set; }
    }
}